pkg/storegateway/series_chunks.go

```diff
@@ -305,32 +305,54 @@ func newPreloadingAndStatsTrackingSetIterator[Set any](ctx context.Context, prel
 }
 
 type loadingSeriesChunksSetIterator struct {
-	chunkReaders  bucketChunkReaders
-	from          seriesChunkRefsSetIterator
-	fromBatchSize int
-	stats         *safeQueryStats
+	chunkReaders           bucketChunkReaders
+	from                   seriesChunkRefsSetIterator
+	fromBatchSize          int
+	stats                  *safeQueryStats
+	ignoreNativeHistograms bool
 
 	current seriesChunksSet
 	err     error
 }
 
-func newLoadingSeriesChunksSetIterator(chunkReaders bucketChunkReaders, from seriesChunkRefsSetIterator, fromBatchSize int, stats *safeQueryStats) *loadingSeriesChunksSetIterator {
+func newLoadingSeriesChunksSetIterator(chunkReaders bucketChunkReaders, from seriesChunkRefsSetIterator, fromBatchSize int, stats *safeQueryStats, ignoreNativeHistograms bool) *loadingSeriesChunksSetIterator {
 	return &loadingSeriesChunksSetIterator{
-		chunkReaders:  chunkReaders,
-		from:          from,
-		fromBatchSize: fromBatchSize,
-		stats:         stats,
+		chunkReaders:           chunkReaders,
+		from:                   from,
+		fromBatchSize:          fromBatchSize,
+		stats:                  stats,
+		ignoreNativeHistograms: ignoreNativeHistograms,
 	}
 }
 
 func (c *loadingSeriesChunksSetIterator) Next() (retHasNext bool) {
-	if c.err != nil {
+	var next *seriesChunksSet
+	tryNext := true
+
+	for tryNext {
+		var err error
+		next, tryNext, err = c.getNext()
+		if err != nil {
+			c.err = err
+			return false
+		}
+	}
+
+	if next == nil {
 		return false
 	}
 
+	c.current = *next
+	return true
+}
+
+func (c *loadingSeriesChunksSetIterator) getNext() (next *seriesChunksSet, tryNext bool, err error) {
+	if c.err != nil {
+		return nil, false, c.err
+	}
```

Comment:
This error is now tracked and assigned mostly in `Next()`, can you also move this if there?

Reply:
Ah yea I think so--removed!